File: Strategy Pattern/Cpp/Duck/MallardDuck.cpp
#include <iostream>
#include "../Behavior/Fly/FlyNoWay.cpp"
#include "../Behavior/Quack/Quack.cpp"
#include "Duck.cpp"

using namespace std;

class MallardDuck : public Duck {
public:
	MallardDuck() {
		flyBehavior = new FlyNoWay();
		quackBehavior = new Quack();
	}

	void display() {
		cout << "I'm Mallard Duck.\n";
	}
};

File: Patterns/Strategy Pattern/CSharp/Duck/MallardDuck.cs
using System;

using Behavior;

namespace Duck 
{
    public class MallardDuck : Duck 
    {
        public MallardDuck() : 
            base(new Behavior.Fly.FlyWithWings(), new Behavior.Quack.Quack()) { }

        public override void OnDisplay() 
        {
            Console.WriteLine("I'm Mallard Duck.");
        }
    }
}